source/tomopy/prep/stripe.py

```diff
@@ -939,3 +941,151 @@ def _remove_stripe_based_interpolation(tomo, snr, size, drop_ratio, norm):
+        tomo = dtype.as_float32(tomo)  # silent convertion to float32 data type
+    out = np.empty_like(tomo, order='C')
+    
+    if tomo.ndim == 3:
+        dz, dy, dx = tomo.shape
+        if (dz == 0) or (dy == 0) or (dx == 0):
+            raise ValueError("The length of one of dimensions is equal to zero")
+    else:
+        raise ValueError("The input array must be a 3D array")
+
+    if size <= 0 or size > dz //  2:
+        raise ValueError("The size of the filter should be larger than zero and smaller than the half of the vertical dimension")
+
+    # perform stripes detection
+    extern.c_stripes_detect3d(np.ascontiguousarray(tomo), 
+                              out, 
+                              size,
+                              radius,
+                              ncore,
+                              dx, dy, dz)
+    return out
+
+
+def stripes_mask3d(weights, 
+                  threshold = 0.6,
+                  min_stripe_length = 20,
+                  min_stripe_depth  = 10,
+                  min_stripe_width = 5,
+                  sensitivity_perc = 85.0,
+                  ncore=None):
+    """
+    Takes the result of the stripes_detect3d module as an input and generates a 
+    binary 3D mask with ones where stripes present. The method tries to eliminate
+    non-stripe features in data by checking the weight consistency in three directions.
+
+    .. versionadded:: 1.14
+
+    Parameters
+    ----------
+    weights : ndarray
+        3D weights array, a result of stripes_detect3d module given in
+        [angles, detY(depth), detX] axis orientation.
+    threshold : float, optional
+        Threshold for the given weights, the smaller values correspond to the stripes
+    min_stripe_length : int, optional
+        Minimum accepted length of a stripe in pixels. Can be large if there are full stripes in the data.
+    min_stripe_depth : int, optional
+        Minimum accepted depth of a stripe in pixels. The stripes do not extend very deep, with this parameter more non-stripe features can be removed. 
+    min_stripe_width : int, optional
+        Minimum accepted width of a stripe in pixels. The stripes can be merged together with this parameter.
+    sensitivity_perc : float, optional
+        The value in percents to impose less strict conditions on length, depth and width of a stripe.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs. All cores will be used
+        if unspecified.
+
+    Returns
+    -------
+    ndarray
+        A binary mask of uint8 data type with stripes highlighted.
+
+    Raises
+    ------
+    ValueError
+        If the input array is not three dimensional.
+
+    """
+    if ncore is None:
+        ncore = mproc.mp.cpu_count()
+
+    input_type = weights.dtype
+    if (input_type != 'float32'):
+        weights = dtype.as_float32(weights)  # silent convertion to float32 data type
+    out = np.uint8(np.empty_like(weights, order='C'))
```

Comment:
I guess you could have convert it to boolean on the wrapper side but sometimes it is easier to do some [morphological processing](https://scikit-image.org/docs/stable/api/skimage.morphology.html) with the mask obtained.

Reply:
Do you mean some morphological processes take non-binary inputs? But this algorithm always returns a binary output?

Reply:
yes, it does always return the binary output. My point here was that for some morphological processing we can feed the binary data output [as is](https://scikit-image.org/docs/stable/auto_examples/applications/plot_morphology.html). I also would like to add a method which takes the binary mask to do the inpainting of the missing data region. So we will need convert boolean back and forth potentially?

Reply:
I'm still confused. The purpose of this thread is to discuss why the current implementation uses `uint8` to store/represent a `bool` value. From my perspective, this makes the data eight times larger than it needs to be. For what reasons did you choose to use `uint8` instead of the `bool` datatype?

You linked the scipy morphology module which contains algorithms which can work with BOTH `bool` AND `uint8`, so I'm not sure what the benefit of using 8 bits to represent a bool mask.

Reply:
OK, I guess I decided not to deal with the conversion to `bool` in the wrapper as C doesn't  have a support for it. Memory-wise we would still need to create `uint8` as an output first before converting it to boolean in the wrapper. If you think we should store it as a boolean - I'm totally fine with it. Shall I implement this?

Reply:
I am asking why the both python and C implementation uses uint8 instead of bool. We want to avoid conversions when possible.

Reply:
ok, I will rewrite to use `bool` instead `uint8` all the way.